drivers/sensor/grow_r502a/grow_r502a.c

```diff
@@ -0,0 +1,697 @@
+	if (ret == FPS_OK) {
+		LOG_INF("Fingerprint Deleted at %d", val->val1);
+	} else {
+		LOG_ERR("Error deleting image 0x%X", ret);
+		return -EIO;
+	}
+
+	ret = fps_get_template_count(dev, &drv_data->template_count);
+	if (ret == FPS_OK) {
+		LOG_DBG("Read success");
+		LOG_INF("Remaining templates count : %d", drv_data->template_count);
+	} else {
+		LOG_ERR("FPS template count get error");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int fps_match(const struct device *dev, struct sensor_value *val)
+{
+	int ret = -1;
+	static uint16_t finger_id, matching_score;
+
+	struct led_params led_ctrl = {
+		.color_code = LED_CTRL_BREATHING,
+		.color = LED_COLOR_BLUE,
+		.speed = LED_SPEED_HALF,
+		.cycle = 0x01,
+	};
+
+	ret = fps_get_image(dev);
+	if (ret == FPS_OK) {
+		ret = fps_led_control(dev, &led_ctrl);
+		if (ret != 0) {
+			return ret;
+		}
+		LOG_DBG("Image taken");
+	} else {
+		LOG_ERR("Error getting image 0x%X", ret);
+		return -EIO;
+	}
+
+	ret = fps_image_to_char(dev, 1);
+	if (ret == FPS_OK) {
+		LOG_DBG("Image converted");
+	} else {
+		LOG_ERR("Error converting image 0x%X", ret);
+		return -EIO;
+	}
+
+	ret = fps_search(dev, 1, &finger_id, &matching_score);
+	if (ret == FPS_OK) {
+		led_ctrl.color_code = LED_CTRL_FLASHING;
+		led_ctrl.color = LED_COLOR_PURPLE;
+		led_ctrl.cycle = 0x01;
+		ret = fps_led_control(dev, &led_ctrl);
+		if (ret != 0) {
+			return ret;
+		}
+		val->val1 = finger_id;
+		val->val2 = matching_score;
+		LOG_INF("Found a matching print! at ID #%d", finger_id);
+	} else if (ret == FPS_NOT_FOUND) {
+		led_ctrl.color_code = LED_CTRL_BREATHING;
+		led_ctrl.color = LED_COLOR_RED;
+		led_ctrl.cycle = 0x02;
+		ret = fps_led_control(dev, &led_ctrl);
+		if (ret != 0) {
+			return ret;
+		}
+		val->val1 = 0;
+		val->val2 = 0;
+		LOG_ERR("Did not find a match");
+	} else {
+		led_ctrl.color_code = LED_CTRL_ON_ALWAYS;
+		led_ctrl.color = LED_COLOR_RED;
+		ret = fps_led_control(dev, &led_ctrl);
+		if (ret != 0) {
+			return ret;
+		}
+		LOG_ERR("Error searching for image 0x%X", ret);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int fps_init(const struct device *dev)
+{
+	int ret;
+	struct led_params led_ctrl = {
+		.color_code = LED_CTRL_FLASHING,
+		.color = LED_COLOR_PURPLE,
+		.speed = LED_SPEED_HALF,
+		.cycle = 0x02,
+	};
+
+	ret = fps_led_control(dev, &led_ctrl);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = fps_verify_password(dev);
+	if (ret == FPS_OK) {
+		LOG_DBG("Correct password, FPS verified");
+	} else {
+		LOG_ERR("Package receive error 0x%X", ret);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int grow_r502a_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct grow_r502a_data *drv_data = dev->data;
+	const struct grow_r502a_config *cfg = dev->config;
+	int ret;
+
+	ret = fps_get_template_count(cfg->bus, &drv_data->template_count);
+	if (ret == FPS_OK) {
+		LOG_DBG("Read success");
+	} else {
+		LOG_ERR("FPS template count get error 0x%X", ret);
+		return -EIO;
+	}
```

Comment:
suggestion, looks like there's a lot of boilerplate here, why not just

```
return fps_get_template_count(cfg->bus, &drv_data->template_count);
```

and let the application log status?

Reply:
thanks for the review, resolved this and all the above mentioned issues.

Reply:
@gmarull Kindly re-review for any issues.Thank You!